### ISSUE TITLE
While using DNSCrypt AdGuard DNS / other encrypted DNS with subscriptions under Split-Tunnel mode and then trying apps at once, ads in apps weren’t blocked. (AG Pro v2.0.2)

### ISSUE THREAD

**Issue by lancelot-moon** [AdguardTeam/AdguardForiOS#740]
Dear Sir,

While using Split-Tunnel mode, DNSCrypt AdGuard DNS and then trying apps at once, ads in apps weren’t blocked.
Video:
https://youtu.be/NoNO1JXhKxE

**After I reviewed DNS Requests Log, it seems that some DNS requests weren’t caught.
I didn't see common ad domains in DNS Requests Log.
So ads in apps appeared.**

Well, ad blocking would be invalid for some time while using DNSCrypt AdGuard DNS.
But ad blocking would be back after trying more some app.
I don't know how to explain.

General AdGuard DNS is ok.
Only DNSCrypt AdGuard DNS has the issue.
Many users met the issue.

You can try General AdGuard DNS / DNSCrypt AdGuard DNS and then open apps with ads by yourself.
ex: The newspaper app:
https://tinyurl.com/y9vb86t5

Compare the difference of DNS Requests Log with General AdGuard DNS / DNSCrypt AdGuard DNS.
In the initial stage, you'll find that common ad domains don't appear in DNS Requests Log while using DNSCrypt AdGuard DNS.
(When using General AdGuard DNS, those common ad domains appear in DNS Requests Log.)

I already sent the bug report to AdGuard Support at 22:32 on 5th April (GMT+8).
   **User ID**: 1425202
   **Subject**: [AdGuard Pro for iOS] bug report

Thanks.

**Comment by X8716e** [comment]
@lancelot-moon , That’s odd. Was the AdGuard SDN subscription the only filter list activated? I’d like to see what occurs when I try.
—
Semi-unrelated question for you: I remember you had been using 1Hosts; if you have stopped using it, may I ask your reasons?

Thank-you.

**Comment by lancelot-moon** [comment]
I already deleted 1hosts for reducing the numbers of rules.

Only 30k+ rules in my test.
Many AG Pro's Taiwanese users met the issue.

**Comment by X8716e** [comment]
Thank-you. So it’s using AdGuard DNSCRYPT servers only, with no subscriptions activated?

**Comment by lancelot-moon** [comment]
> Only 31k+ rules (AG SDN filter and other filters) in subscriptions for my test.
31k+ rules (AG SDN filter and other filters) and AdGuard DNSCRYPT server... = =

**Comment by X8716e** [comment]
So it’s a doubling of the same rules, essentially, yes?

**Comment by lancelot-moon** [comment]
> So it’s a doubling of the same rules, essentially, yes?

It's not important.
Some DNS requests didn't appear in log when using AdGuard DNSCRYPT server, which is more important.
Simply try.

**Comment by X8716e** [comment]
If AdGuard’s DNSCRYPT severs are used with AdGuard’s SDN subscription, it is important since it’s essentially like using the same filter list twice over. Lancelot-moon, I believe your issue true for you and the “many users” you mention. I am not disputing that. I am, however, theorising that issue may be somewhat more complicated.

**Comment by lancelot-moon** [comment]
I’m tired to explain for you since I already said it’s not important.

I additionally deleted AG SDN filter from my subscriptions and then tried.

@Ivanlin
In the video:
https://youtu.be/F9fKuGveqbA

When using general AG DNS, the ads in app are blocked.
But ads in app appear when using DNSCrypt AG DNS.

**Comment by X8716e** [comment]
My mistake for engaging in conversation regarding this issue. I won’t make it again.

**Comment by IvanIin** [comment]
@lancelot-moon @X8716e 
Yes. it's a bug. The tunnel starts processing dns requests too early, when the dnscrypt is not fully initialized. And in split mode, ios switches to system dns for 1 minute. A minute later, as a rule, it starts sending requests through our tunnel again and everything starts working as it should.

**Comment by lancelot-moon** [comment]
Thank you that you prove that it's a bug, otherwise X8716e doesn't believe.
I often suggest AdGuard Pro to Taiwanese netfriends on PTT BBS, which helps AdGuard Pro to be in a high rank on app store.

Many Taiwanese netfriends didn't give enough information about some issues.
The common 2 issues:
According to my observation...
1. Unable to connect to AdGuard VPN.
a. They maybe also met the issue, https://github.com/AdguardTeam/AdguardForiOS/issues/657.
b. They subscribed too many rules, which let VPN tunnel be broken down.

2. Ad blocking is invalid.
a. They used DNScrypt AdGuard DNS or other encrypted DNS servers with subscriptions.
b. They subscribed too many rules, which let VPN tunnel approach collapse and occurred some abnormal situation.

**Comment by lancelot-moon** [comment]
>Yes. it's a bug. The tunnel starts processing dns requests too early, when the dnscrypt is not fully initialized. And in split mode, ios switches to system dns for 1 minute. A minute later, as a rule, it starts sending requests through our tunnel again and everything starts working as it should.

Yes, it seems that there is no this issue if using Full-Tunnel mode.
Ad blocking can normally work when internet connection is ok.

But there are other 2 issues after shifting from Split-Tunnel mode to Full-Tunnel mode & using DNSCrypt AdGuard DNS.

https://github.com/AdguardTeam/AdguardForiOS/issues/743
https://github.com/AdguardTeam/AdguardForiOS/issues/742

**Comment by camelot-sun** [comment]
@X8716e apologies if lancelot moon offended you. I understand and i know you are just trying to help. As a software developer, we need facts. We need the why’s to help explain and solve an issue.

@lancelot-moon , as a fellow asian, and as a developer, you are a bit disrespectful to your fellow tester. He is just trying to help. He asks questions to debug a potential issue and to formluate solutions in your issue.

You probably haven’t work in an IT company or is part of an IT team that’s why your responce towards him is different.

For a technical person to help you, he will ask many questions. This questions are all relevant. You might think it’s garbage, but it’s not. This questions will help us undertand the situation and formulate a solution.

Your video does not say anything on how the bug was established. It just says what the bug is, but not the WHY and HOW.

**Comment by zebrum** [comment]
let's have one issue #744

**Comment by lancelot-moon** [comment]
> You probably haven’t work in an IT company or is part of an IT team that’s why your responce towards him is different.

I ever worked in IC design house and system manufacturer.
Only a negligible engineer.

> Your video does not say anything on how the bug was established. It just says what the bug is, but not the WHY and HOW.

That's not true.
I had written more detailed test conditions in my article as possible.
Especially, I had emphasized:
> I had told that you should note the difference of DNS Requests Log with General AdGuard DNS / DNSCrypt AdGuard DNS when opening the same app with ads.

But he insists about the unimportant thing (the same filters in AdGuard DNS & SDN filter) for several times, which let me feel tired.
Will ad blocking be invalid with local SDN filter and general AdGuard DNS in general?

> Please don’t boast about how you contribute the app to be high ranking in taiwan. AdGuard app is already popular. Even without you, it can still make it to number 1....

I talked about helping AdGuard on BBS.
We know that users are often lazy to report issues to official team because they think it's trouble.
Although I'm not AdGuard's staff, Taiwanese users report issues to me, not to AdGuard Support.
It's because I often introduced AdGuard on PTT BBS.
Not only I but also many other users met the issue.
Let AdGuard team know this issue is NOT a rare case.

And then maybe you're a leader in your team in an IT company.
But you don't understand market.
I'm the first person who introduced AdGuard Pro for iOS to netfriends on social network in Taiwan.
I helped AdGuard Pro for iOS to be in high ranking, which is a fact.
If I didn't introduce AdGuard Pro for iOS on PTT BBS, AdGuard had lower reputation for Taiwanese iOS users.
On PTT BBS, there are 56 articles about AdBlock by Future Mind.
There are only 16 articles about AdGuard Pro for iOS on PTT BBS.
(9/16 articles are written by me.)
Actually, the most popular No.1 adblocker for iOS in Taiwan is AdBlock by Future Mind.

I translate AdGuard, become a tester and introduce AdGuard to Taiwanese netfriends for free.
I didn't get any valued benefits from AdGuard.
Simply, AdGuard is the greatest adblocker and worth to share the information with Taiwanese in my thought, but it doesn't represent it can be in high ranking on app store.
AdBlock by Future Mind usually won AdGuard Pro for iOS on Taiwan app store.

I will also not reply you anymore in this topic.

Hope that AdGuard team can smoothly solve issues.
